enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,340 @@
+
+This metric will give us insight into user experiences operating the Machine Health
+Check. For example, users might think they are utilizing the Machine Health
+Check correctly only to find out that it has been misconfigured and not
+performing its role. If users have a large cluster with the health check
+enabled, but not covering any nodes, this might be an opportunity to reach out
+and see if we can help with their configuration. This can also help us identify
+areas where increased education will have value.
+
+**Failed Node Cleanups**
+
+Epics: Proactive Support
+
+This metric will help us identify conditions where the Cluster Autoscaler is
+silently cleaning up failied node creations. We should trigger an alert if the
+autoscaler cleans up more than X hosts in Y time period. This alert would be
+useful for users that might not realize they’re having these infrastructure problems.
+This is another case where signals across multiple clusters might indicate
+that a cloud or network area is having problems.
+
+For example, our internal continuous integration infrastructure team had a
+situation where two MachineSets had been configured for autoscaling.
+One MachineSet would scale up without issue, then the other would not. While
+this was happening, the Machines would fail, alarms would fire, and the autoscaler
+would dispose of the nodes that failed to come online. These failed nodes were
+being silently cleaned up by the autoscaler.
+
+**Total Scale Events**
+
+Epics: Demonstration of Value
+
+This metric will help us to show the level of activity that the Cluster
+Autoscaler is performing. Collecting data on the total number of scale up and
+scale down events, possibly as two individual metrics, we could correlate with
+instance pricing to demonstrate how much using the autoscaler saves our users.
+This can also serve as a good proxy for how many users are actually using the
+autoscaler in a meaningful way, not just who has turned it on.
+
+**Unable to Scale Events**
+
+Epics: Proactive Support
+
+This metric will help us identify situations when the Cluster Autoscaler is
+unable to perform scaling operations. To maximize the value of this data, we
+will want to report categories based on the different types of failures.
+Events related to min/max threshold on MachineSets probably don’t need
+alerting, but the inability to scale for other reasons (eg, hitting max CPU threshold or
+memory, but haven’t hit max node count) are useful to collect. These events can
+show opportunities to inform users about resource limitations in their clusters.
+
+**Machines with Old Delete Timestamp**
+
+Epics: Proactive Support
+
+This metric can show us stale resources that exist within clusters. By collecting
+data for machines that are stuck in a deletion phase for prolonged periods of time
+we can identify opportunities to inform users and create alerts around these
+stale resources. This alerting might need to be tunable (if possible), but a
```

Comment:
Alerts cannot be tunable. You can have different levels of severity. So for example per threshold you can have severity info, higher threshold can have severity critical which would mean someone needs to be paged.

Reply:
ack, i can remove the language about tunable alerts.